providers/implementations/kem/ml_kem.c

```diff
@@ -111,18 +118,25 @@ static int mlkem_decapsulate_init(void *vctx, void *vkey,
 static int mlkem_set_ctx_params(void *vctx, const OSSL_PARAM params[])
 {
     PROV_MLKEM_CTX *ctx = (PROV_MLKEM_CTX *)vctx;
+    const OSSL_PARAM *p;
 
     debug_print("MLKEMKEM set ctx params %p\n", ctx);
     if (ctx == NULL)
         return 0;
     if (params == NULL)
         return 1;
 
+    if ((p = OSSL_PARAM_locate_const(params, OSSL_KEM_PARAM_MLKEM_ENC_ENTROPY)) != NULL
```

Comment:
Design question: Is this parameter OK to make available for general use? Or is there some mechanism I didn't spot that only makes this active during testing?

Reply:
I didn't use any mechanism to make this available only for testing (i.e. it's available for general use). But as whether to make this available to general use, I guess that's up to us. As far as I can tell, other providers don't usually hide params like this. However, BoringSSL doesn't make it available for general use (plus some other functions as shown [here](https://boringssl.googlesource.com/boringssl/+/refs/heads/master/crypto/mlkem/internal.h)).

Reply:
I'm also undecided on that -- let's try to resolve that as part of the documentation as discussed.

Reply:
Now spent some time on this: FIPS 203 section 3.3 IMO is pretty clear on these parameters: "The interfaces for these functions should not be made available to applications other than for testing purposes".

"Funny" of course is that violating this is the basis for https://datatracker.ietf.org/doc/draft-ietf-lamps-kyber-certificates/06/. I'm a bit wary wading into this: How far is OpenSSL willing to go supporting draft specs that are not final, @mattcaswell @t8m?

How can we make these params "test-time only" until that is sorted, @andrewkdinh ?

Reply:
@baentsch you might find this presentation particularly interesting, about this topic:

- https://youtu.be/uETYyGwD3gA?si=M-_k7HphNJksu5hi&t=2221
- https://datatracker.ietf.org/meeting/121/materials/slides-121-pquip-fips-issues-with-deploying-ml-kem-and-ml-dsa-04

Reply:
"The interfaces for these functions should not be made available to applications other than for testing purposes" -  this is something to be solved by using FIPS indicators. We do not need to resolve it now. Just make these available as normal OSSL_PARAMs.

Reply:
From [the presentation I linked above](https://datatracker.ietf.org/meeting/121/materials/slides-121-pquip-fips-issues-with-deploying-ml-kem-and-ml-dsa-04):

![image](https://github.com/user-attachments/assets/8182548c-8837-4d56-b25f-38392cca9982)


@t8m the consensus among implementers right now can be summarized as:

![image](https://github.com/user-attachments/assets/d64db1af-11a4-40d7-b033-6960375363de)

Reply:
For the other issues emerged when interpreting FIPS 203 & co, this is a good overall summary:

![image](https://github.com/user-attachments/assets/b0672334-4005-4893-a525-f6f00d506b3f)

Reply:
I agree with @t8m. This isn't a problem for now. Solve it later. It might mean we make this kind of interface available in the default provider, but not the FIPS provider.

Reply:
OK, I'll then document these params. Besides, a "should not" is not a "must not". Thanks for linking the presentation @romen .